vm/src/vm/vm_ops.rs

```diff
@@ -219,13 +268,53 @@ impl VirtualMachine {
     }
 
     pub fn _mul(&self, a: &PyObject, b: &PyObject) -> PyResult {
-        self.call_or_reflection(
-            a,
-            b,
-            identifier!(self, __mul__),
-            identifier!(self, __rmul__),
-            |vm, a, b| Err(vm.new_unsupported_binop_error(a, b, "*")),
-        )
+        let ret = self.binary_op1(a, b, PyNumberMethodsOffset::Multiply);
+
+        // for sequence protocol
+        if let Ok(ref r) = ret {
+            if r.rich_compare_bool(
+                self.ctx.not_implemented.as_object(),
+                PyComparisonOp::Eq,
+                self,
+            )? {
+                let seq_a = PySequence::new(a, self);
+                let seq_b = PySequence::new(b, self);
+
+                // TODO: remove the duplicated conditions
+                // The reason I don't put the below sepecial condition into `PyBytes::AsSequence::repeat` is
+                // we cannot use `to_owned()` in `PyBytes::repeat`.
```

Comment:
you can fix `PyBytes::repeat` to take `PyRef<Self>` or `&Py<Self>` instead of `&self` to allow `to_owned()`